sklearn/model_selection/_search.py

```diff
@@ -918,15 +1005,32 @@ class RandomizedSearchCV(BaseSearchCV):
 
     Attributes
     ----------
-    grid_scores_ : list of named tuples
-        Contains scores for all parameter combinations in param_grid.
-        Each entry corresponds to one parameter setting.
-        Each named tuple has the attributes:
-
-            * ``parameters``, a dict of parameter settings
-            * ``mean_validation_score``, the mean score over the
-              cross-validation folds
-            * ``cv_validation_scores``, the list of scores for each fold
+    search_results_ : dict of numpy (masked) ndarrays
```

Comment:
I think @jnothman is right, especially because this is likely to be an issue at the end of a long fit.  I guess `grid_scores_` should raise an exception if (when) multiple metrics are used.  They're not "angry users", if anything, they will be users who embark in the process of updating their projects to keep up with scikit-learn changes. We want to make it smooth, such that they can stop in the middle of the rewriting effort, if they have to, and things will still work, right?

Maybe in such a case we could use a shorter deprecation cycle, if you think that would help keep things cleaner.